[CI] DataTiersMigrationsTests.testIndexDataTierMigration fails periodically

**Build scan**:
- [7.9.3] https://gradle-enterprise.elastic.co/s/e6moxgn6emvvc
- [7.10.0] https://gradle-enterprise.elastic.co/s/ctax63vbhxdxc
- https://gradle-enterprise.elastic.co/s/zza4pg6ncbpcq

**Repro line**:
```
REPRODUCE WITH: ./gradlew ':x-pack:plugin:ilm:internalClusterTest' --tests "org.elasticsearch.xpack.ilm.DataTiersMigrationsTests.testIndexDataTierMigration" \
  -Dtests.seed=323EED72404458A3 \
  -Dtests.security.manager=true \
  -Dtests.locale=de-CH \
  -Dtests.timezone=Europe/Prague \
  -Druntime.java=11 \
  -Dtests.fips.enabled=true
```
**Reproduces locally?**: No

**Applicable branches**: master

**Failure history**:
- [7.9.3] https://gradle-enterprise.elastic.co/s/hkntbxhn4uzrq  
- [7.10.0] https://gradle-enterprise.elastic.co/s/3uwwc7j6ml3e6 
- https://gradle-enterprise.elastic.co/s/m4eits6zzubhu

**Failure excerpt**:

```
java.lang.AssertionError: 
Expected: is "check-migration"
     but: was "migrate"
	at __randomizedtesting.SeedInfo.seed([323EED72404458A3:8EBC766D3CC57784]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.xpack.ilm.DataTiersMigrationsTests.lambda$testIndexDataTierMigration$1(DataTiersMigrationsTests.java:141)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:934)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:907)
	at org.elasticsearch.xpack.ilm.DataTiersMigrationsTests.testIndexDataTierMigration(DataTiersMigrationsTests.java:134)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
	at java.base/java.lang.Thread.run(Thread.java:834)
	Suppressed: java.lang.AssertionError: 
Expected: is "cold"
     but: was "warm"
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ilm.DataTiersMigrationsTests.lambda$testIndexDataTierMigration$1(DataTiersMigrationsTests.java:140)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:922)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: is "cold"
     but: was "warm"
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ilm.DataTiersMigrationsTests.lambda$testIndexDataTierMigration$1(DataTiersMigrationsTests.java:140)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:922)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: is "cold"
     but: was "warm"
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ilm.DataTiersMigrationsTests.lambda$testIndexDataTierMigration$1(DataTiersMigrationsTests.java:140)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:922)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: is "cold"
     but: was "warm"
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ilm.DataTiersMigrationsTests.lambda$testIndexDataTierMigration$1(DataTiersMigrationsTests.java:140)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:922)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: is "cold"
     but: was "warm"
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ilm.DataTiersMigrationsTests.lambda$testIndexDataTierMigration$1(DataTiersMigrationsTests.java:140)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:922)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: is "cold"
     but: was "warm"
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ilm.DataTiersMigrationsTests.lambda$testIndexDataTierMigration$1(DataTiersMigrationsTests.java:140)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:922)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: is "cold"
     but: was "warm"
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ilm.DataTiersMigrationsTests.lambda$testIndexDataTierMigration$1(DataTiersMigrationsTests.java:140)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:922)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: is "cold"
     but: was "warm"
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ilm.DataTiersMigrationsTests.lambda$testIndexDataTierMigration$1(DataTiersMigrationsTests.java:140)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:922)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: is "cold"
     but: was "warm"
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ilm.DataTiersMigrationsTests.lambda$testIndexDataTierMigration$1(DataTiersMigrationsTests.java:140)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:922)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: is "cold"
     but: was "warm"
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ilm.DataTiersMigrationsTests.lambda$testIndexDataTierMigration$1(DataTiersMigrationsTests.java:140)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:922)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: is "cold"
     but: was "warm"
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ilm.DataTiersMigrationsTests.lambda$testIndexDataTierMigration$1(DataTiersMigrationsTests.java:140)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:922)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: is "cold"
     but: was "warm"
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ilm.DataTiersMigrationsTests.lambda$testIndexDataTierMigration$1(DataTiersMigrationsTests.java:140)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:922)
		... 39 more
	Suppressed: java.lang.AssertionError: 
Expected: is "cold"
     but: was "warm"
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.xpack.ilm.DataTiersMigrationsTests.lambda$testIndexDataTierMigration$1(DataTiersMigrationsTests.java:140)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:922)
		... 39 more
```

It failed only twice in a week, so I am not going to mute it for now.

--- a/x-pack/plugin/ilm/src/internalClusterTest/java/org/elasticsearch/xpack/ilm/DataTiersMigrationsTests.java
+++ b/x-pack/plugin/ilm/src/internalClusterTest/java/org/elasticsearch/xpack/ilm/DataTiersMigrationsTests.java
@@ -120,7 +120,6 @@ public class DataTiersMigrationsTests extends ESIntegTestCase {
         assertTrue(res.isAcknowledged());
 
         assertBusy(() -> {
-            client().admin().cluster().prepareReroute().get();
             ExplainLifecycleRequest explainRequest = new ExplainLifecycleRequest().indices(managedIndex);
             ExplainLifecycleResponse explainResponse = client().execute(ExplainLifecycleAction.INSTANCE,
                 explainRequest).get();
@@ -133,7 +132,6 @@ public class DataTiersMigrationsTests extends ESIntegTestCase {
         logger.info("starting a warm data node");
         internalCluster().startNode(warmNode(Settings.EMPTY));
         assertBusy(() -> {
-            client().admin().cluster().prepareReroute().get();
             ExplainLifecycleRequest explainRequest = new ExplainLifecycleRequest().indices(managedIndex);
             ExplainLifecycleResponse explainResponse = client().execute(ExplainLifecycleAction.INSTANCE,
                 explainRequest).get();
@@ -148,7 +146,6 @@ public class DataTiersMigrationsTests extends ESIntegTestCase {
 
         // wait for lifecycle to complete in the cold phase after the index has been migrated to the cold node
         assertBusy(() -> {
-            client().admin().cluster().prepareReroute().get();
             ExplainLifecycleRequest explainRequest = new ExplainLifecycleRequest().indices(managedIndex);
             ExplainLifecycleResponse explainResponse = client().execute(ExplainLifecycleAction.INSTANCE,
                 explainRequest).get();
@@ -184,7 +181,6 @@ public class DataTiersMigrationsTests extends ESIntegTestCase {
         assertTrue(res.isAcknowledged());
 
         assertBusy(() -> {
-            client().admin().cluster().prepareReroute().get();
             ExplainLifecycleRequest explainRequest = new ExplainLifecycleRequest().indices(managedIndex);
             ExplainLifecycleResponse explainResponse = client().execute(ExplainLifecycleAction.INSTANCE,
                 explainRequest).get();
@@ -208,7 +204,6 @@ public class DataTiersMigrationsTests extends ESIntegTestCase {
         //  1. start another cold node so both the primary and replica can relocate to the cold nodes
         //  2. remove the tier routing setting from the index again (we're doing this below)
         assertBusy(() -> {
-            client().admin().cluster().prepareReroute().get();
             ExplainLifecycleRequest explainRequest = new ExplainLifecycleRequest().indices(managedIndex);
             ExplainLifecycleResponse explainResponse = client().execute(ExplainLifecycleAction.INSTANCE,
                 explainRequest).get();
@@ -223,7 +218,6 @@ public class DataTiersMigrationsTests extends ESIntegTestCase {
 
         // wait for lifecycle to complete in the cold phase
         assertBusy(() -> {
-            client().admin().cluster().prepareReroute().get();
             ExplainLifecycleRequest explainRequest = new ExplainLifecycleRequest().indices(managedIndex);
             ExplainLifecycleResponse explainResponse = client().execute(ExplainLifecycleAction.INSTANCE,
                 explainRequest).get();

--- a/x-pack/plugin/ilm/src/internalClusterTest/java/org/elasticsearch/xpack/ilm/DataTiersMigrationsTests.java
+++ b/x-pack/plugin/ilm/src/internalClusterTest/java/org/elasticsearch/xpack/ilm/DataTiersMigrationsTests.java
@@ -120,6 +120,7 @@ public class DataTiersMigrationsTests extends ESIntegTestCase {
         assertTrue(res.isAcknowledged());
 
         assertBusy(() -> {
+            client().admin().cluster().prepareReroute().get();
             ExplainLifecycleRequest explainRequest = new ExplainLifecycleRequest().indices(managedIndex);
             ExplainLifecycleResponse explainResponse = client().execute(ExplainLifecycleAction.INSTANCE,
                 explainRequest).get();
@@ -132,6 +133,7 @@ public class DataTiersMigrationsTests extends ESIntegTestCase {
         logger.info("starting a warm data node");
         internalCluster().startNode(warmNode(Settings.EMPTY));
         assertBusy(() -> {
+            client().admin().cluster().prepareReroute().get();
             ExplainLifecycleRequest explainRequest = new ExplainLifecycleRequest().indices(managedIndex);
             ExplainLifecycleResponse explainResponse = client().execute(ExplainLifecycleAction.INSTANCE,
                 explainRequest).get();
@@ -146,6 +148,7 @@ public class DataTiersMigrationsTests extends ESIntegTestCase {
 
         // wait for lifecycle to complete in the cold phase after the index has been migrated to the cold node
         assertBusy(() -> {
+            client().admin().cluster().prepareReroute().get();
             ExplainLifecycleRequest explainRequest = new ExplainLifecycleRequest().indices(managedIndex);
             ExplainLifecycleResponse explainResponse = client().execute(ExplainLifecycleAction.INSTANCE,
                 explainRequest).get();
@@ -181,6 +184,7 @@ public class DataTiersMigrationsTests extends ESIntegTestCase {
         assertTrue(res.isAcknowledged());
 
         assertBusy(() -> {
+            client().admin().cluster().prepareReroute().get();
             ExplainLifecycleRequest explainRequest = new ExplainLifecycleRequest().indices(managedIndex);
             ExplainLifecycleResponse explainResponse = client().execute(ExplainLifecycleAction.INSTANCE,
                 explainRequest).get();
@@ -204,6 +208,7 @@ public class DataTiersMigrationsTests extends ESIntegTestCase {
         //  1. start another cold node so both the primary and replica can relocate to the cold nodes
         //  2. remove the tier routing setting from the index again (we're doing this below)
         assertBusy(() -> {
+            client().admin().cluster().prepareReroute().get();
             ExplainLifecycleRequest explainRequest = new ExplainLifecycleRequest().indices(managedIndex);
             ExplainLifecycleResponse explainResponse = client().execute(ExplainLifecycleAction.INSTANCE,
                 explainRequest).get();
@@ -218,6 +223,7 @@ public class DataTiersMigrationsTests extends ESIntegTestCase {
 
         // wait for lifecycle to complete in the cold phase
         assertBusy(() -> {
+            client().admin().cluster().prepareReroute().get();
             ExplainLifecycleRequest explainRequest = new ExplainLifecycleRequest().indices(managedIndex);
             ExplainLifecycleResponse explainResponse = client().execute(ExplainLifecycleAction.INSTANCE,
                 explainRequest).get();

--- a/x-pack/plugin/ilm/src/internalClusterTest/java/org/elasticsearch/xpack/ilm/DataTiersMigrationsTests.java
+++ b/x-pack/plugin/ilm/src/internalClusterTest/java/org/elasticsearch/xpack/ilm/DataTiersMigrationsTests.java
@@ -128,7 +128,7 @@ public class DataTiersMigrationsTests extends ESIntegTestCase {
             IndexLifecycleExplainResponse indexLifecycleExplainResponse = explainResponse.getIndexResponses().get(managedIndex);
             assertThat(indexLifecycleExplainResponse.getPhase(), is("warm"));
             assertThat(indexLifecycleExplainResponse.getStep(), is(DataTierMigrationRoutedStep.NAME));
-        });
+        }, 30, TimeUnit.SECONDS);
 
         logger.info("starting a warm data node");
         internalCluster().startNode(warmNode(Settings.EMPTY));
@@ -141,7 +141,7 @@ public class DataTiersMigrationsTests extends ESIntegTestCase {
             IndexLifecycleExplainResponse indexLifecycleExplainResponse = explainResponse.getIndexResponses().get(managedIndex);
             assertThat(indexLifecycleExplainResponse.getPhase(), is("cold"));
             assertThat(indexLifecycleExplainResponse.getStep(), is(DataTierMigrationRoutedStep.NAME));
-        });
+        }, 30, TimeUnit.SECONDS);
 
         logger.info("starting a cold data node");
         internalCluster().startNode(coldNode(Settings.EMPTY));
@@ -156,7 +156,7 @@ public class DataTiersMigrationsTests extends ESIntegTestCase {
             IndexLifecycleExplainResponse indexLifecycleExplainResponse = explainResponse.getIndexResponses().get(managedIndex);
             assertThat(indexLifecycleExplainResponse.getPhase(), is("cold"));
             assertThat(indexLifecycleExplainResponse.getStep(), is("complete"));
-        });
+        }, 30, TimeUnit.SECONDS);
     }
 
     public void testUserOptsOutOfTierMigration() throws Exception {

